common4j/src/main/com/microsoft/identity/common/java/providers/oauth2/AuthorizationRequest.java

```diff
@@ -308,7 +312,15 @@ public String toString() {
     public URI getAuthorizationRequestAsHttpRequest() throws ClientException {
         try {
             final CommonURIBuilder builder = new CommonURIBuilder(getAuthorizationEndpoint());
-            builder.addParametersIfAbsent(ObjectMapper.serializeObjectHashMap(this));
+            final Map<String, Object> serializedMap = ObjectMapper.serializeObjectHashMap(this);
+
+            // If the flight is not enabled, do not include these fields in the query parameter map
+            if (!CommonFlightsManager.INSTANCE.getFlightsProvider().isFlightEnabled(CommonFlight.ENABLE_AM_API_WORKPROFILE_EXTRA_QUERY_PARAMETERS)) {
+                serializedMap.remove(MicrosoftAuthorizationRequest.WP_AVAILABLE_EXTRA_PARAMETER_NAME);
+                serializedMap.remove(Device.PlatformIdParameters.MANUFACTURER);
```

Comment:
It seems like the fields were added and we're removing them? If the flight is disabled then why we added the fields in the first place?

Reply:
The fields are marked as exposed in `MicrosoftAuthorizationRequest.java`, so that they will be picked up by the `ObjectMapper.serializeObjectHashMap(this)` call in this method. They will always be included in the map as they are fields in the class itself, so checking for flight here and removing them is our way to make sure they don't get attached to the request if the flight is false. To my knowledge there isn't a straightforward way to conditionally assign an annotation. This is the next best thing in my opinion.

Reply:
I reworked it to not annotate them as exposed, but i had to mark them transient to avoid serialization. This seems to re-order them at the end of the query string, but i think this is fine.